File: ui_recording/app/src/main/java/edu/illinois/recordingservice/ScreenShot.java
package edu.illinois.recordingservice;

import android.graphics.Bitmap;
import android.graphics.Rect;

import java.util.ArrayList;

public class ScreenShot {
    public static final int TYPE_CLICK = 0;
    public static final int TYPE_SCROLL = 1;
    public static final int TYPE_LONG_CLICK = 2;

    private Bitmap bitmap;
    private Rect rect;
    private int action_type;

    private ArrayList<Rect> vh;

    public ScreenShot(Bitmap bitmap, Rect rect, int action_type, ArrayList<Rect> vh) {
        this.bitmap = bitmap;
        this.rect = rect;
        this.action_type = action_type;
        this.vh = vh;
    }


    public Bitmap getBitmap() {
        return bitmap;
    }

    public void setBitmap(Bitmap bitmap) {
        this.bitmap = bitmap;
    }

    public Rect getRect() {
        return rect;
    }

    public void setRect(Rect rect) {
        this.rect = rect;
    }

    public int getAction_type() {
        return action_type;
    }

    public void setAction_type(int action_type) {
        this.action_type = action_type;
    }

    public ArrayList<Rect> getVh() {
        return vh;
    }

    public void setVh(ArrayList<Rect> vh) {
        this.vh = vh;
    }
}

File: app/src/main/java/edu/illinois/odim/ScreenShot.kt
package edu.illinois.odim

import android.graphics.Bitmap
import android.graphics.Rect

class ScreenShot(
    var bitmap: Bitmap?,
    var rect: Rect?,
    action_type: Int,
    scroll_coords: Pair<Int, Int>?,
    var vh: ArrayList<Rect>?
) {
    companion object {
        const val TYPE_CLICK = 0
        const val TYPE_SCROLL = 1
        const val TYPE_LONG_CLICK = 2
        const val TYPE_SELECT = 3
    }

    var actionType : Int = action_type

    var scrollCoords : Pair<Int, Int>? = scroll_coords  // (x scroll delta, y scroll delta)

}